bigquery/google/cloud/bigquery/job.py

```diff
@@ -931,13 +972,35 @@ def __init__(self):
     https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.copy.writeDisposition
     """
 
+    @property
+    def kms_key_name(self):
+        """str: Resource ID of Cloud KMS key
+
+        Resource ID of Cloud KMS key to encrypt destination table or ``None``
+        if using default encryption.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.copy.destinationEncryptionConfiguration.kmsKeyName
+        """
+        return self._kms_key_name
+
+    @kms_key_name.setter
+    def kms_key_name(self, value):
+        if not isinstance(value, six.string_types) and value is not None:
```

Comment:
@tswast do we do this kind of validation in other BigQuery stuff? We should generally avoid doing excessive type-checking or validation in the clients.

Reply:
We used to do more and have been removing them as they've caused problems. Agreed that this check isn't necessary.

Reply:
Removed and refactored.